### PR TITLE
Add editor pattern test in BracketedPatternTest

### PR DESCRIPTION
**Pull request by chiao26168**
```[tasklist]
### Compulsory checks
- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [ ] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.
```

The existing test class, org.jabref.logic.citationkeypattern.BracketedPatternTest.java, currently lacks coverage for editor patterns. To address this gap, I have added a new test method, testEditorFieldMarkers() to ensure that the editor pattern functions correctly.

During the execution of testEditorFieldMarkers(), one test case failed. The input provided was "'New', '[edtr3_1]', 'Isaac Newton'", and the expected output was 'New'. However, the actual output was null.

Screen Shot 2023-05-23 at 7 01 36 AM

(output of the failed test case)

The source of the error can be found in org.jabref.logic.citationkeypattern.BracketedPattern. 
As stated in [Customize the citation key generator](https://docs.jabref.org/setup/citationkeypatterns), when the pattern matches "edtrN_M", the citation key should consist of the first N characters of the Mth editor's last name. 

Here is a screenshot highlighting the problem in the source code:
Screen Shot 2023-05-23 at 7 10 10 AM

To fix this issue, the following return statement should be used

```
return authNofMth(editorList,
                            Integer.parseInt(nums[0]),
                            Integer.parseInt(nums[1]));
```

### CODE REVIEW

**Comment by Siedlerchr** [comment]
Thank you very much! Please go ahead and fix this as well :)

**Review by k3KAW8Pnf7mkmdSMPHz27 — CHANGES_REQUESTED** [review]
Overall this looks very good, and I love the detailed description! I am slightly disturbed that the `authNofMth` error seems to have been here since 2017, so test cases for this are definitively needed :scream: 

Remaining
- I am not sure about one of the test cases. Could you write down your thoughts regarding it, or if we have a similar test case in a different part of the code, reference it? I am hesitant to include it because I believe it might be a misuse of the editors' field, in which case all bets are off, and there isn't a right or wrong behavior of JabRef
- Would you mind fixing the code style check complaints? It is much less important in a test case, but since it is only complaining about new lines
- Since the behavior is changing, please add an entry to the change list, even if it is now the correct behavior as described in the documentation

File: src/test/java/org/jabref/logic/citationkeypattern/BracketedPatternTest.java [hunk 1]
```diff
@@ -660,4 +660,29 @@ void expandBracketsWithoutProtectiveBracesUsingUnprotectTermsModifier() {
...
+            "'Ne', '[edtrIni2]', 'Isaac Newton'",
+            "'New', '[edtr3]', 'Isaac Newton'",
+            "'Newton', '[edtr7]', 'Isaac Newton'",
+            "'New', '[edtr3_1]', 'Isaac Newton'",
+            "'Newton.Maxwell', '[edtr.edtr.ea]', 'Isaac Newton and James Maxwell'",
+            "'Newton', '[edtrshort]', 'Isaac Newton'",
+            "'Newton', '[editorLast]', 'Isaac Newton'",
+            "'I', '[editorLastForeIni]', 'Isaac Newton'",
+
+            "'Agency', '[editors]', 'European Union Aviation Safety Agency'",
```
[the remark this answers is not in the document]
Comment (Siedlerchr) [reply src/test/java/org/jabref/logic/citationkeypattern/BracketedPatternTest.java:678]: Not malformed, but it does not make sense to have a corporate author without curly brackets-

[the remark this answers is not in the document]
Comment (k3KAW8Pnf7mkmdSMPHz27) [reply src/test/java/org/jabref/logic/citationkeypattern/BracketedPatternTest.java:678]: @Siedlerchr you agree that this one particular test should be removed?

File: src/test/java/org/jabref/logic/citationkeypattern/BracketedPatternTest.java [hunk 2]
```diff
@@ -660,4 +660,28 @@ void expandBracketsWithoutProtectiveBracesUsingUnprotectTermsModifier() {
...
```
Comment (Siedlerchr) [inline src/test/java/org/jabref/logic/citationkeypattern/BracketedPatternTest.java:678]:
```suggestion

```

**Comment by Siedlerchr** [comment]
@chiao26168  can you please add a changelog entry?

**Comment by chiao26168** [comment]
> @chiao26168 can you please add a changelog entry?

Should I add two entries, one for adding the test case and the other for fixing the error?

**Comment by Siedlerchr** [comment]
Changelog is intended for the end users so just one for the changed behavior/fix

File: CHANGELOG.md [hunk 3]
```diff
@@ -93,6 +93,7 @@ Note that this project **does not** adhere to [Semantic Versioning](http://semve
 - We fixed an issue where no preview could be generated for some entry types and led to an exception [#9947](https://github.com/JabRef/jabref/issues/9947)
 - We fixed an issue where the Linux terminal working directory argument was malformed and therefore ignored upon opening a terminal [#9953](https://github.com/JabRef/jabref/issues/9953)
 - We fixed an issue where an Automatic Keyword Group could not be deleted in the UI. [#9778](https://github.com/JabRef/jabref/issues/9778)
+- We fixed an issue where the citation key pattern `[edtrN_M]` was not handled properly. [#9946](https://github.com/JabRef/jabref/pull/9946)
```
Comment (k3KAW8Pnf7mkmdSMPHz27) [inline CHANGELOG.md:96]:
```suggestion
- We fixed an issue where the citation key pattern `[edtrN_M]` returned the wrong editor. [#9946](https://github.com/JabRef/jabref/pull/9946)
```

Nitpick/Opinionated: Explains it a bit more and is easier to understand. The PR is accepted from my point of view anyway.

**Review by k3KAW8Pnf7mkmdSMPHz27 — APPROVED** [review]
LGTM! We greatly appreciate the detailed description, the missing test-cases, detection of a bug, and fixing of the bug :sweat_smile:

**Review by Siedlerchr — APPROVED** [review]

**Comment by Siedlerchr** [comment]
Thanks again for the tests and finding these bugs!

**Pull request merged**